new-lamassu-admin/src/lamassu/App.js

```diff
@@ -70,11 +71,36 @@ const useStyles = makeStyles({
   }
 })
 
+const GET_USER_DATA = gql`
+  query userData {
+    userData {
+      id
+      username
+      role
+      enabled
+      last_accessed
+      last_accessed_from
+      last_accessed_address
+    }
+  }
+`
+
 const Main = () => {
   const classes = useStyles()
   const location = useLocation()
   const history = useHistory()
-  const { wizardTested } = useContext(AppContext)
+  const { wizardTested, userData, setUserData } = useContext(AppContext)
+
+  const [getUserData, { loading }] = useLazyQuery(GET_USER_DATA, {
+    onCompleted: userResponse => {
+      if (!userData && userResponse?.userData)
+        setUserData(userResponse.userData)
+    }
+  })
+
+  useEffect(() => {
+    getUserData()
+  }, [getUserData])
```

Comment:
Hmm, I don't think you need the effect at all, won't this execute the same amount of times as if it was a regular query? I'm not 100% sure but it feels strange.